doc/wikis/cereal.md

```diff
@@ -0,0 +1,204 @@
+## SHOGUN `Cereal` serialization framework
+
+#### Table of Contents
+
+- [Motivation](#motivation)
+- [For SHOGUN developers](#For SHOGUN developers)
+  - [Examples] (#Examples)
+- [For serialization framework developers] (#For serialization framework developers)
+  - [Serialization interface] (#Serialization interface)
+  - [Serialization methods in `SGObject`] (#Serialization methods in `SGObject`)
+  - [Serialization methods in `Any`] (#Serialization methods in `Any`)
+  - [Serialization methods in `SGVector`, `SGMatirx` and `SGReferencedData`] (#Serialization methods in `SGVector`, `SGMatirx` and `SGReferencedData`)
+
+
+### Motivation
+
+[`Cereal`](http://uscilab.github.io/cereal/) is a header-only C++11 serialization library that is fast, light-weight, and easy to extend. 
+
+The `Cereal` serialization framework in SHOGUN uses the new tag parameter framework, which allows the easy and readable archive of `SGObject` data.
+
+### For SHOGUN developers
+
+- `Cereal` serialization library is required for SHOGUN compilation. If no `Cereal` is found, SHOGUN will automatically download the library to `third_party/`.
+
+- SHOGUN now supports the serialization of data into 3 formats: binary, XML, and JSON archives. The 3 pairs of save/load methods can be called by:
+```
+save_binary(filename);
+save_json(filename);
+save_xml(filename);
+load_binary(filename);
+load_json(filename);
+load_xml(filename);
+```
+
+- All parameters saved in tag parameter list for one `SGObject` can be saved and load by:
+
+```
+SGObject obj_save;
+obj_save.save_json(filename);
+
+SGObject obj_load;
+obj_load.load_json(filename);
+```
+
+- Customized archives can be added as shown [here](http://uscilab.github.io/cereal/serialization_archives.html)
+
+#### Examples
+
+ `CCerealObject` class defined in `tests/unit/io/CerealObject.h` is a `SGObject`-based class used for `Cereal` serialization unit tests. 
+ We also use `CCerealObject` here to show how to serialize `SGObject` in SHOGUN.
+ 
+ In `CCerealObject`, we initialized a member `SGVector<float64_t> m_vector` and regisitered it to the parameter list in constructors:
+ 
+ ```
+ #include <shogun/base/SGObject.h>
+ #include <shogun/lib/SGVector.h>
+ 
+ using namespace shogun;
+ 
+ class CCerealObject : public CSGObject
+ {
+     public:
+        // Construct CCerealObject from input SGVector
+        CCerealObject(SGVector<float64_t> vec) : CSGObject()
+        {
+            m_vector = vec;
+            init_params();
+        }
+        
+        // Default constructor
+        CCerealObject() : CSGObject()
+        {
+            m_vector = SGVector<float64_t>(5);
+            m_vector.set_const(0);
+            init_params();
+        }
+    
+        const char* get_name() const { return "CerealObject"; }
+        
+     protected:
+        // Register m_vector to parameter list with name(tag) "test_vector"
+        void init_params()
+        {
+
+            register_param("test_vector", m_vector);
```

Comment:
use `register_member("test_vector", &m_vector);` as m_vector is a member variable. See [tags readme](https://github.com/sanuj/shogun/blob/tag_readme/doc/tags_param.md#register_param--register_member).